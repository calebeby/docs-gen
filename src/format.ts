import prettier from 'prettier'

const options = {
  semi: false,
  parser: 'typescript',
}

const prefix = 'type Data = '

export const format = (text: string) => prettier.format(prefix + text, options)
