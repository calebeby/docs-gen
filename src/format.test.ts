import { format } from './format'

test('format', () => {
  expect(
    format(
      '{ hello?: "hello" | 40, goodbye: {data: ((string | false | {name: "hi" | "bye"})[])}[] }',
    ),
  ).toMatchSnapshot()
})
