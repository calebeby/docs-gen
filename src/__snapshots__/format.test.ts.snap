// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format 1`] = `
"type Data = {
  hello?: \\"hello\\" | 40
  goodbye: { data: (string | false | { name: \\"hi\\" | \\"bye\\" })[] }[]
}
"
`;
